File: crates/dewasm-backend-python/units/wasi/environ_sizes_get.py
# requires: memory/iws
def wasi_environ_sizes_get(self, count_ptr, buf_size_ptr):
    self.memory.iws(count_ptr, len(self.env))
    self.memory.iws(buf_size_ptr, sum(len(e) + 1 for e in self.env))
    return self.ERRNO_SUCCESS

File: crates/dewasm-backend-perl/units/wasi/environ_sizes_get.pl
# requires: memory/i32_store
sub wasi_environ_sizes_get {
    my ($self, $count_ptr, $buf_size_ptr) = @_;
    my $bytes = 0;
    $bytes += length($_) + 1 for @{$self->{env}};
    $self->{memory}->i32_store($count_ptr, scalar @{$self->{env}});
    $self->{memory}->i32_store($buf_size_ptr, $bytes);
    return ERRNO_SUCCESS;
}
